enarx-keep-sgx-shim/src/handler.rs

```diff
@@ -644,4 +644,28 @@ impl<'a> Handler<'a> {
 
         trusted.len() as u64
     }
+
+    // Do clock_gettime syscall
+    pub fn clock_gettime(&mut self) -> u64 {
+        self.trace("clock_gettime", 2);
+
+        let clk_id = self.aex.gpr.rdi.raw();
+        let ts = self.aex.gpr.rsi.raw() as *mut libc::timespec;
+
+        unsafe {
+            let (_, untrusted, _) = self.block.buf.align_to::<libc::timespec>();
+            let uaddr = untrusted.as_ptr();
+            let result = self.proxy(request!(libc::SYS_clock_gettime => clk_id, uaddr));
```

Comment:
Since you wrote this, we added the `Cursor` type which should help with allocations like this.

```suggestion
            let cursor = self.block.cursor();
            let mut untrusted = cursor.copy_slice(&[*ts]).or(Err(libc::EINVAL))?;
            let result = self.proxy(request!(libc::SYS_clock_gettime => clk_id, &mut untrusted[0]))?;
            if result[0] != 0 { self.attacked(); }
            *ts = untrusted[0];
            Ok(result)
```